server/controllers/recall/returnToCustodyDateController.ts

```diff
@@ -1,43 +1,123 @@
+import { AdjustmentDto } from '../../@types/adjustmentsApi/adjustmentsApiTypes'
 
 export default class ReturnToCustodyDateController extends RecallBaseController {
   validateFields(req: FormWizard.Request, res: Response, callback: (errors: unknown) => void) {
     super.validateFields(req, res, errors => {
       const { values } = req.form
       const recallDate = getRecallDate(req)
+      const rtcDate = new Date(values.returnToCustodyDate as string)
 
       /* eslint-disable-next-line  @typescript-eslint/no-explicit-any */
       const validationErrors: any = {}
 
-      if (values.inPrisonAtRecall === 'false' && isBefore(values.returnToCustodyDate as string, recallDate)) {
-        validationErrors.returnToCustodyDate = this.formError('returnToCustodyDate', 'mustBeEqualOrAfterRecallDate')
-      }
+      if (values.inPrisonAtRecall === 'false') {
+        if (isBefore(values.returnToCustodyDate as string, recallDate)) {
+          validationErrors.returnToCustodyDate = this.formError('returnToCustodyDate', 'mustBeEqualOrAfterRecallDate')
+        }
+
+        const existingAdjustments: AdjustmentDto[] = getExistingAdjustments(req)
+        // We want to check that any overlapping UAL here is a recall UAL, otherwise fail validation per RCLL-322
+
+        const conflictingAdjustments: AdjustmentDto[] = this.getConflictingAdjustments(
+          recallDate,
+          rtcDate,
+          existingAdjustments,
+        )
 
+        if (
+          conflictingAdjustments.some(
+            adjustment => this.isNonUalAdjustment(adjustment) || this.isNonRecallUal(adjustment),
+          )
+        ) {
+          validationErrors.returnToCustodyDate = this.formError('returnToCustodyDate', 'conflictingAdjustment')
+        }
+      }
       callback({ ...errors, ...validationErrors })
     })
   }
 
+  isNonUalAdjustment(adjustment: AdjustmentDto) {
+    return adjustment.adjustmentType !== 'UNLAWFULLY_AT_LARGE'
+  }
+
+  isNonRecallUal(adjustment: AdjustmentDto) {
+    return adjustment.adjustmentType === 'UNLAWFULLY_AT_LARGE' && adjustment.unlawfullyAtLarge!.type !== 'RECALL'
+  }
+
+  getConflictingAdjustments(recallDate: Date, rtcDate: Date, searchResults?: AdjustmentDto[]): AdjustmentDto[] {
+    if (!searchResults || searchResults.length === 0) {
+      return []
+    }
+
+    const conflictingAdjustments = searchResults.filter(adjustment =>
+      this.doesConflict(recallDate, rtcDate, adjustment),
+    )
+
+    return conflictingAdjustments
+  }
+
+  doesConflict(recallDate: Date, rtcDate: Date, adjustment: AdjustmentDto): boolean {
+    if (!adjustment.fromDate || !adjustment.toDate) {
+      return false
+    }
+    const recallStart = recallDate.getTime()
+    const recallEnd = rtcDate.getTime()
+    const adjStart = parseISO(adjustment.fromDate)
+    const adjEnd = parseISO(adjustment.toDate)
+
+    const startsBeforeRecallEnds = isBefore(adjStart, recallEnd)
+    const endsAfterRecallStarts = isAfter(adjEnd, recallStart)
```

Comment:
I think we're missing some conditions here that this method for the time being doesn't care about the distinction between but will need to return true regardless of the manner in which they conflict. Will pick up with you in Slack